# 1. Number of Unique Words

### A.1.1 Opening and Reading The Text 


To count the number of unique words, first, we should display the text.

In [1]:
#openning the text file 
mytext = open('latin_text.txt')

#reading the file 
mytext = mytext.read() 

#display the text
print(mytext) 

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec in erat maximus, eleifend ante eget, feugiat libero. Maecenas vel pharetra orci. Morbi vitae finibus augue. Nunc pharetra ac lacus vel efficitur. Cras sed urna non ex luctus imperdiet. Sed eu volutpat nunc. Maecenas et nunc tellus. Nunc eget purus consectetur, ultricies leo iaculis, gravida elit. Nunc molestie tortor dolor, nec laoreet lectus sollicitudin sed. Duis ultricies at diam nec tincidunt. Donec at metus placerat lacus commodo molestie nec ut risus. Curabitur fermentum justo lectus, non maximus mauris placerat vel. Fusce auctor ex massa, nec condimentum enim interdum nec. Maecenas consectetur varius nunc, auctor pellentesque mi feugiat eu. Cras lacinia, ipsum vitae malesuada eleifend, mi magna hendrerit enim, id fringilla urna ante in magna.

Suspendisse dolor nisl, interdum sit amet rutrum eget, congue sit amet odio. Vestibulum vestibulum orci sed volutpat facilisis. Quisque consectetur convallis mauris. In pretium

### A.1.2. Cleaning the All Punctuations From text

Second, we should delete the all punctuations.

In [2]:
mytext = mytext.replace('\n',' ')  # deleting \n 
mytext = mytext.replace(',',' ')   #deleting ,
mytext = mytext.replace('.',' ')   #deleting .
print(mytext)

Lorem ipsum dolor sit amet  consectetur adipiscing elit  Donec in erat maximus  eleifend ante eget  feugiat libero  Maecenas vel pharetra orci  Morbi vitae finibus augue  Nunc pharetra ac lacus vel efficitur  Cras sed urna non ex luctus imperdiet  Sed eu volutpat nunc  Maecenas et nunc tellus  Nunc eget purus consectetur  ultricies leo iaculis  gravida elit  Nunc molestie tortor dolor  nec laoreet lectus sollicitudin sed  Duis ultricies at diam nec tincidunt  Donec at metus placerat lacus commodo molestie nec ut risus  Curabitur fermentum justo lectus  non maximus mauris placerat vel  Fusce auctor ex massa  nec condimentum enim interdum nec  Maecenas consectetur varius nunc  auctor pellentesque mi feugiat eu  Cras lacinia  ipsum vitae malesuada eleifend  mi magna hendrerit enim  id fringilla urna ante in magna   Suspendisse dolor nisl  interdum sit amet rutrum eget  congue sit amet odio  Vestibulum vestibulum orci sed volutpat facilisis  Quisque consectetur convallis mauris  In pretium

### A.1.3. Counting the Unique Words of the Text

Now, we should find how many unique words exist in the text.

First, we would define a fuction which counts the unique words in the text.

While I am defining a function, first, I would make lowercas all lettrs of a word in the text.

That's because, Python is sensitive to capital letters. 

For example.  'lorem' and 'Lorem' would be counted as different two words. 

For this reason, I would set() function to eliminate repititon of words.

In [3]:
#defining a function which counts words
def unique_words(the_text):
    #splitting the test into words
    allwords_of_text = the_text.split() 
    
    #translating the all letters in the text into lowercase 
    lower_text = [] 
    for x in allwords_of_text:
        lower_text.append(x.lower())
    
    #eleminating the repeated words
    b =set(lower_text)
    
    #number of unique words
    return len(b)  

Secondly, we would apply this function to our text to find unique words.

In [4]:
#application of unique words function to random_text.txt
unique_words(mytext)

151

So, number of unique words of the text is 151. 